### PR TITLE
Add check for env variables presence

### PR DESCRIPTION
**Pull request by ondrejsevcik**
Should solve #44

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,23 +23,37 @@ module.exports = {
...
 
     if (fs.existsSync(configFactory)) {
       Object.assign(options, require(configFactory)(this._resolveEnvironment()));
     }
 
-    if (fs.existsSync(options.path) && dotenv.config({ path: options.path })) {
-      let loadedConfig = dotenv.parse(fs.readFileSync(options.path));
-      let allowedKeys = options.clientAllowedKeys || [];
+    let loadedConfig = dotenv.config({ path: options.path });
+    if (loadedConfig.error && options.failOnMissingKey) {
```
Comment (SergeAstapov) [inline index.js]: @ondrejsevcik these two errors does not seem to be related - `loadedConfig.error` would mean that `.env` file was not found or has syntax error. What did you intend to achieve with this change?

File: index.js [hunk 2]
```diff
@@ -23,23 +23,37 @@ module.exports = {
...
 
-      this._config = allowedKeys.reduce((accumulator, key) => {
-        accumulator[key] = loadedConfig[key];
+    let allowedKeys = options.clientAllowedKeys || [];
 
-        return accumulator;
-      }, {});
+    if (options.failOnMissingKey) {
+      allowedKeys.map(key => {
+        if (!loadedConfig.parsed[key]) {
```
Comment (SergeAstapov) [inline index.js]: @ondrejsevcik this can break your build if key value is `false` but I believe there are use cases when `false` is a valid key. Should we compare with `undefined`?

File: index.js (hunk 1 above)
Comment (jasonmit) [reply index.js]: I was also confused, can you drop a comment in the source if it's meaningful?

**Comment by ondrejsevcik** [comment]
thanks for the review, hopefully it's much clearer now. And I've also incorporated your suggestions.

File: index.js [hunk 3]
```diff
@@ -23,23 +23,45 @@ module.exports = {
...
+        throw new Error(loadingErrMsg);
+      } else {
+        console.warn(loadingErrMsg);
+        return;
+      }
     }
+
+    let allowedKeys = options.clientAllowedKeys || [];
+
+    allowedKeys.map(key => {
```
Comment (SergeAstapov) [inline index.js]: @ondrejsevcik minor - could you please use `forEach` here as it seems more appropriate?

Comment (ondrejsevcik) [reply index.js]: true

**Comment by SergeAstapov** [comment]
LGTM!
@jasonmit any suggestions?

Thank you @ondrejsevcik for working on this!

**Comment by jasonmit** [comment]
LGTM, thanks!
